delft3dworker/views.py

```diff
@@ -74,16 +83,134 @@ class ScenarioViewSet(viewsets.ModelViewSet):
     API endpoint that allows scenarios to be viewed or edited.
     """
     serializer_class = ScenarioSerializer
-    filter_backends = (filters.DjangoObjectPermissionsFilter,
-                       filters.OrderingFilter)
+    filter_backends = (
+        filters.DjangoFilterBackend,
+        filters.SearchFilter,
+        filters.OrderingFilter,
+        filters.DjangoObjectPermissionsFilter,
+    )
     permission_classes = (permissions.IsAuthenticated,
                           ViewObjectPermissions,)
 
     # Default ordering by id, so latest run is last in list
     # reverse by setting ('-id',)
     ordering = ('id',)
 
-    queryset = Scenario.objects.all()
+    # Our own custom filter to create custom search fields
+    # this creates &name= among others
+    filter_class = ScenarioFilter
+    search_fields = ('^name', )
+
+    queryset = Scenario.objects.none()
+
+    def get_queryset(self):
+        """
+        Optionally restricts the returned purchases to a given parameter,
+        by filtering against a `parameter` query parameter in the URL.
+
+        Possible values:
+            # filters on key occurance
+            - parameter="parameter
+            # filters on key occurance and value
+            - parameter="parameter,value"
+            # filters on key occurance and value between min & max
+            - parameter="parameter,minvalue,maxvalue"
+        """
+        queryset = Scenario.objects.all()
+
+        # Filter on parameter
+        parameters = self.request.query_params.getlist('parameter', [])
+        template = self.request.query_params.getlist('template', [])
+
+        if len(parameters) > 0:
```

Comment:
We cannot return when parameters are not there: at the end we still need to check for template as well.